### PR TITLE
fix: update health check endpoint to ensure all dbs are operational

### PR DESCRIPTION
**Pull request by schenedx**
[MST-1231](https://openedx.atlassian.net/browse/MST-1231)
Update the health check endpoint to check all important databases: default and (reports OR reports_v1).
If any of the dbs are not accessible, the overall status of the service will be `UNAVAILABLE`.

@edx/masters-devs-cosmonauts Please review

### CODE REVIEW

File: analyticsdataserver/views.py [hunk 1]
```diff
@@ -71,36 +71,55 @@ class HealthView(APIView):
...
+        overall_status = self.STATUS_UNAVAILABLE
+
+        # First try the default database.
+        # The default database hosts user and auth info. Without it, no API would function
+
+        default_db_status = self._get_connection_status('default')
+        analytics_db_status = self._get_connection_status(
+            getattr(settings, 'ANALYTICS_DATABASE')
+        )
+        analytics_v1_db_status = self._get_connection_status(
```
Comment (ashultz0) [inline analyticsdataserver/views.py:106]: it should probably only do this if the system cares about the v1 DB, i.e. settings.DATA_API_V1_ENABLED = True

Comment (schenedx) [reply analyticsdataserver/views.py:106]: @ashultz0 There is no such setting `DATA_API_V1_ENABLED`. That setting is instead defined in the [Insights repo](https://github.com/openedx/edx-analytics-dashboard/search?q=DATA_API_V1_ENABLED)

Comment (Zacharis278) [reply analyticsdataserver/views.py:106]: I imagine we don't want this to fail for existing systems w/ no v1 DB. Should we maybe have a different 'v1' healthcheck endpoint if we need to rely on the error response code?

File: analyticsdataserver/settings/test.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 }
 
 ANALYTICS_DATABASE = 'analytics'
-ANALYTICS_DATABASE_V1 = 'analytics_v1'
+ANALYTICS_DATABASE_V1 = None
```
Comment (ashultz0) [inline analyticsdataserver/settings/test.py:36]: this might fail the router tests, not sure

File: analyticsdataserver/views.py [hunk 3]
```diff
@@ -71,36 +71,55 @@ class HealthView(APIView):
...
+
+        # First try the default database.
+        # The default database hosts user and auth info. Without it, no API would function
+
+        default_db_status = self._get_connection_status('default')
+
+        analytics_db_name = getattr(settings, 'ANALYTICS_DATABASE_V1')
+        if not analytics_db_name:
+            # If the v1 db is not defined, we use the default analytics db name setting instead
+            analytics_db_name = getattr(settings, 'ANALYTICS_DATABASE')
```
Comment (ashultz0) [inline analyticsdataserver/views.py:107]: while we are transitioning we need both to be defined to be healthy

and there are enterprise tables in the old analytics db that we aren't planning to move. So I think this should be if v1 check it, and always check v0

File: analyticsdataserver/settings/test.py (hunk 2 above)
Comment (schenedx) [reply analyticsdataserver/settings/test.py:36]: I checked and it does not

**Review by ashultz0 — APPROVED** [review]

**Review by Zacharis278 — APPROVED** [review]

**Pull request merged**